### PR TITLE
ZK-4115: listbox + frozen column widths resetting when scrolling

### PR DESCRIPTION
**Pull request by charlesqqq**

### CODE REVIEW

File: zul/src/archive/web/js/zul/mesh/Frozen.js [hunk 1]
```diff
@@ -314,16 +314,19 @@ zul.mesh.Frozen = zk.$extends(zul.Widget, {
...
+				ftcells = ftrows ? ftrows.rows[0].cells : null,
+				cellsWidth = [];
+
+			for (var i = 0; i < totalCols; i++)
+				cellsWidth.push(jq.px0(hdcells[i].offsetWidth));
 
 			for (var faker, i = 0; hdcol && i < totalCols; hdcol = hdcol.nextSibling, i++) {
-				if (hdcol.style.width.indexOf('px') == -1) {
-					var sw = hdcol.style.width = jq.px0(hdcells[i].offsetWidth),
+				if (hdcol.style.width.indexOf('px') == -1 || this._syncFaker) {
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/mesh/Frozen.js]: 1. any timing to set `this._syncFaker` false?
2. or simplify the condition to always recalculate

File: zul/src/archive/web/js/zul/mesh/MeshWidget.js [hunk 2]
```diff
@@ -1291,16 +1291,73 @@ zul.mesh.MeshWidget = zk.$extends(zul.Widget, {
...
+		if (newTotalWidth == oldTotalWidth)
+			return;
+
+		for (var i = 0, header = head.firstChild, hdcol = this.ehdfaker.firstChild; i < totalCols;
+				header = header.nextSibling, hdcol = hdcol.nextSibling, i++) {
+			if (header.getWidth() || header.getHflex())
+				continue;
+
+			var fakerStyleWidth = hdcol.style.width,
+				multiple = newTotalWidth / oldTotalWidth,
```
Comment (charlesqqq) [inline zul/src/archive/web/js/zul/mesh/MeshWidget.js]: multiplier

File: zul/src/archive/web/js/zul/mesh/MeshWidget.js [hunk 3]
```diff
@@ -1291,16 +1291,73 @@ zul.mesh.MeshWidget = zk.$extends(zul.Widget, {
...
+			width0 = zul.mesh.MeshWidget.WIDTH0,
+			newTotalWidth = parseFloat(this.ebodytbl.style.width),
+			oldTotalWidth = 0;
+		for (var i = 0, header = head.firstChild, hdcol = this.ehdfaker.firstChild; i < totalCols;
+				header = header.nextSibling, hdcol = hdcol.nextSibling, i++) {
+			var fakerStyleWidth = hdcol.style.width;
+			if (fakerStyleWidth == '')
+				return;
+
+			var isHiding = fakerStyleWidth == width0,
```
Comment (charlesqqq) [inline zul/src/archive/web/js/zul/mesh/MeshWidget.js]: isHidden

**Comment by scribetw** [comment]
@zkadmin rebase

**Pull request merged**